### ISSUE TITLE
How to use vue-instantsearch in SSR with Vuex?

### ISSUE THREAD

**Issue by rentorm** [algolia/vue-instantsearch#455]
I'm struggling to integrate vue-instantsearch with Vuex store in Vue SSR app.

I've tried to follow https://github.com/algolia/vue-instantsearch-examples/tree/master/examples/ssr however this example is using only `context.store` and to adapt it using Vuex

My integartion attempst are following:

```
<template>
  <div class="vwp-single">
    <ais-index :searchStore="searchStore" :auto-search="false">
      <ais-search-box placeholder="Find products"/>
      <ais-refinement-list attribute-name="colors"></ais-refinement-list>
      <ais-results>
        <template scope="{ result }">
          <div>
            <ais-highlight :result="result" attribute-name="name"></ais-highlight>
          </div>
        </template>
      </ais-results>
    </ais-index>
    <div class="clearfix"></div>
  </div>
</template>

<script>
import {
  createFromAlgoliaCredentials,
  createFromSerialized,
  FACET_OR
} from 'vue-instantsearch'

import { mapGetters } from 'vuex'

const fetchInitialData = (store, route) => {
  let store1
  store1 = createFromAlgoliaCredentials(
    'latency',
    '6be0576ff61c053d5f9a3225e2a90f76'
  )
  store1.indexName = 'bestbuy'
  store1.query = route.params.query ? route.params.query : ''
  store1.addFacet('colors', FACET_OR)
  store1.highlightPreTag = '<mark>'
  store1.highlightPostTag = '</mark>'
  store1.start()
  store1.refresh()
  return store1.waitUntilInSync().then(() => {
    store.dispatch(`pt/searchStore`, store1.serialize())
  })
}

export default {
  computed: {
    ...mapGetters('pt', ['searchStore'])
  },
  prefetch: fetchInitialData,
  beforeMount () {
    if (!window.__INITIAL_STATE__) {
      throw new Error('Not state was found.')
    }
    this.searchStore = createFromSerialized(
      window.__INITIAL_STATE__.pt.searchStore
    )
  },
  methods: {
    loadResults () {
      fetchInitialData(this.$store, this.$route)
    }
  },
  created () {
    this.loadResults()
  },
  watch: {
    '$route' () {
      this.searchStore.query = this.$route.params.query
        ? this.$route.params.query
        : ''
    },
    'searchStore.query' (to) {
      if (to.length === 0) {
        this.$router.push({ name: 'map' })
      } else {
        this.$router.push({ name: 'mapSearch', params: { query: to } })
      }
    }
  }

}
</script>
```
if I remove `ais-index` and just render out `{{ searchStore }}` I can see data returned, but if I try to mount it on ais-index component, it fails with following errors:

![screen shot 2018-05-28 at 11 48 22](https://user-images.githubusercontent.com/1099350/40612259-aec68236-6271-11e8-8019-b8da0a9ee304.png)

Would much appreciate if someone could point me in right direction hiow to debug this or show example code how to integrate vue-instantsearch with Vuex and SSR

**Comment by Haroenv** [comment]
Hey! Would it be possible to set this up in a complete GitHub repo, there's still quite a few things that could be the source of this issue, so it'll be easier to debug like that!

**Comment by rentorm** [comment]
I've set Github repo with current implementation, hope this helps - https://github.com/rentorm/vue-instantsearch-vuex

I think there might be something with a way `this.searchStore = createFromSerialized` methods are being used to set `searchStore` but I can't figure out another way how to initiate it on server.

**Comment by Haroenv** [comment]
I'm still trying some things out, but I'm mostly wondering _why_ you want the `searchStore` to be in Vuex, would it not also be fine to do it like in the example?

**Comment by Haroenv** [comment]
The reason errors are happening is because `searchStore` only has getters in your Vuex store (`...mapGetters`), but no setters (since that would make the need for Vuex alltogether fade away (no longer unidirectional). For that reason I suggest using context directly (like in the example) 😊

**Comment by rentorm** [comment]
reason I'm trying to set up with Vuex is because whole application state is managed through `store` rather than `context` and I did not want to have a set up where they both are managed separately.

I made a quick try for them both to coexist in this branch - https://github.com/rentorm/vue-instantsearch-vuex/tree/using-context and it does generate SSR values, but those still are not passed to `context.state`

I guess I'm just missing something fundamental here... 😄

**Comment by rentorm** [comment]
I did not find a way to mix `context.state` with `store.state` and moved `vue-instantsearch` code in store `action`. I created new brach with this code - https://github.com/rentorm/vue-instantsearch-vuex/tree/vuex-action

This now will render search results on `Search` page while navigating from other pages. However if page is requested directly search results are rendered by server, I get console error:

`vue-instantsearch.esm.js:854 Uncaught (in promise) TypeError: this._localSearchStore.enableCache is not a function` 

and whole search/filder functionality is broken. Also no longer is possible to navigate to other pages.

I feel I'm so close (I hope) 😊
